modules/account-lib/src/coin/sol/utils.ts

```diff
@@ -0,0 +1,86 @@
+import { Keypair, PublicKey } from '@solana/web3.js';
+import bs58 from 'bs58';
+import BigNumber from 'bignumber.js';
+
+const ENCODED_BLOCK_HASH_LENGTH = 44;
+const DECODED_BLOCK_HASH_LENGTH = 32;
+const ENCODED_ADDRESS_LENGTH = 44;
+const DECODED_ADDRESS_LENGTH = 32;
+const DECODED_SIGNATURE_LENGTH = 64;
+
+/** @inheritdoc */
+export function isValidAddress(address: string): boolean {
+  try {
+    return (
+      !!address &&
+      new RegExp('[1-9A-HJ-NP-Za-km-z]').test(address) &&
+      address.length === ENCODED_ADDRESS_LENGTH &&
+      bs58.decode(address).length === DECODED_ADDRESS_LENGTH
+    );
+  } catch (e) {
+    return false;
+  }
+}
+
+/** @inheritdoc */
+export function isValidBlockId(hash: string): boolean {
+  return (
+    !!hash &&
+    new RegExp('[1-9A-HJ-NP-Za-km-z]').test(hash) &&
+    hash.length === ENCODED_BLOCK_HASH_LENGTH &&
+    bs58.decode(hash).length === DECODED_BLOCK_HASH_LENGTH
+  );
+}
+
+/** @inheritdoc */
+export function isValidPrivateKey(prvKey: string | Uint8Array): boolean {
+  const key: Uint8Array = typeof prvKey === 'string' ? this.base58ToUint8Array(prvKey) : prvKey;
+  try {
+    return !!Keypair.fromSecretKey(key);
+  } catch (e) {
+    return false;
+  }
+}
+
+/** @inheritdoc */
+export function isValidPublicKey(pubKey: string): boolean {
+  try {
+    return (
+      !!pubKey &&
+      new RegExp('[1-9A-HJ-NP-Za-km-z]{32,44}').test(pubKey) &&
+      PublicKey.isOnCurve(new PublicKey(pubKey).toBuffer())
+    );
+  } catch (e) {
+    return false;
+  }
+}
+
+/** @inheritdoc */
+export function isValidSignature(signature: string): boolean {
+  return !!signature && bs58.decode(signature).length === DECODED_SIGNATURE_LENGTH;
+}
+
+/** @inheritdoc */
+// TransactionId are the first signature on a Transaction
+export function isValidTransactionId(txId: string): boolean {
+  return this.isValidSignature(txId);
+}
+
+/**
+ * Returns whether or not the string is a valid amount number
```

Comment:
should mention that amounts are expected to be in lamports (or small units for tokens)